apps/prairielearn/elements/pl-excalidraw/pl-excalidraw.py

```diff
@@ -0,0 +1,100 @@
+import base64
+import json
+from pathlib import Path
+
+import chevron
+import lxml.html
+import prairielearn as pl
+
+
+def prepare(element_html: str, _: pl.QuestionData) -> None:
+    element = lxml.html.fragment_fromstring(element_html)
+    pl.check_attribs(element, ["name"], ["width", "height", "file", "file_dir"])
+
+
+def render(element_html: str, data: pl.QuestionData) -> str:
+    element = lxml.html.fragment_fromstring(element_html)
+    with open("pl-excalidraw.mustache", "r", encoding="utf-8") as template:
+        drawing_name = pl.get_string_attrib(element, "name")
+        initial_content: str = ""
+
+        def load_file_content() -> str:
```

Comment:
I don't think this should be handled as an inner function, you should define this outside of `render` and pass in all the required arguments.

Reply:
I'm not using it elsewhere so I don't think it's a good idea to move it outside.

Reply:
It's not really about reuse, I've generally avoided using inner functions like this because they lead to more indentation in your code and make it harder to read.

Reply:
I see, I'll hoist it